### PR TITLE
batch inference and data loader unfinished

### PR DESCRIPTION
**Pull request by sjkoelle**
This contains batch-inference enabling methods in several of the model.Tacotron2 classes.  In data_loader it also contains unfinished drafts of multipurpose TTSDataset, Collate, and Batch classes, as well as prepare sequence methods from uberduck-py for facilitating inference with the site.

### CODE REVIEW

File: uberduck_ml_dev/models/tacotron2.py [hunk 1]
```diff
@@ -550,23 +568,32 @@ def forward(self, x, input_lengths):
...
 
         outputs, _ = nn.utils.rnn.pad_packed_sequence(outputs, batch_first=True)
         return outputs
 
-    def inference(self, x):
+    def inference(self, x, input_lengths):
+        device = x.device
         for conv in self.convolutions:
-            x = F.dropout(F.relu(conv(x)), self.dropout_rate, self.training)
+            x = F.dropout(F.relu(conv(x.to(device))), 0.5, self.training)
```
Comment (zachwe) [inline uberduck_ml_dev/models/tacotron2.py:584]: is this line necessary? `device` is defined as `x.device`, so `x.to(device)` is redundant, no?

Also, why the change to hard-code the dropout rate?

File: uberduck_ml_dev/models/tacotron2.py [hunk 2]
```diff
@@ -713,15 +740,18 @@ def forward(self, inputs):
...
 
         encoder_outputs = torch.cat((embedded_text,), dim=2)
 
-        mel_outputs, gate_outputs, alignments = self.decoder.inference(encoder_outputs)
-
+        memory_lengths = input_lengths
+        mel_outputs, gate_outputs, alignments = self.decoder.inference(
+            encoder_outputs, memory_lengths
+        )
+        # mel_outputs, gate_outputs, alignments, mel_lengths = self.decoder.inference(encoder_outputs, memory_lengths)
```
Comment (zachwe) [inline uberduck_ml_dev/models/tacotron2.py:754]: I think we need `mel_lengths`. Otherwise users of this function will have to look at `gate_outputs` to figure out how long each batch is, which seems wasteful.

**Review by zachwe — APPROVED** [review]
LGTM to marge into the larger feature branch as discussed

**Comment by zachwe** [comment]
This will go to master via a different branch

File: uberduck_ml_dev/models/tacotron2.py (hunk 1 above)
Comment (sjkoelle) [reply uberduck_ml_dev/models/tacotron2.py:584]: wut, i dont remember making this change at all

File: uberduck_ml_dev/models/tacotron2.py (hunk 2 above)
Comment (sjkoelle) [reply uberduck_ml_dev/models/tacotron2.py:754]: where should mel_lengths be called from?
